sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala

```diff
@@ -193,6 +196,9 @@ class SQLBuilder(logicalPlan: LogicalPlan) extends Logging {
     case OneRowRelation =>
       ""
 
+    case Hint(_, _, child) =>
```

Comment:
I remember the reason. Originally, I designed `Hint` is removed by some rules of `Analyzer` or `Optimizer`. But, now, it's changed. Only rules of `Analyzer` handles that and removes all unknown hints. `CheckAnalysis` is the correct position.